concent_api/api_testing_common.py

```diff
@@ -163,33 +162,32 @@ def _print_data(data, url):
     validate_content_type(response.headers['Content-Type'], expected_content_type)
     validate_response_message(response.content, expected_message_type, private_key, public_key)
     print()
-    if response.status_code in [202, 204]:
+    content_type = response.headers['Content-Type']
+    if 'text/html' in content_type:
         return None
-    else:
+    elif 'json' in content_type:
+        return json.loads(response._content)
+    elif 'application/octet-stream' in content_type:
         return try_to_decode_golem_message(private_key, public_key, response.content)
+    else:
+        raise UnexpectedResponse(f'Unexpected response content_type. Responses content type is {content_type}.')
 
 
 def _print_response(private_key, public_key, response):
     if response.content is None:
         print('RESPONSE: <empty>')
-    elif len(response.content) != 0:
-        _print_message_from_response(private_key, public_key, response)
     else:
-        print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
-        if response.text not in ['', None]:
-            print('RAW RESPONSE (text): {}'.format(response.text))
-
-
-def _print_message_from_response(private_key, public_key, response):
-    print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
-    print('MESSAGE:')
-    print('Concent-Golem-Messages-Version = {}'.format(response.headers['concent-golem-messages-version']))
-    if response.headers['Content-Type'] == 'application/octet-stream':
-        _print_message_from_stream(private_key, public_key, response.content)
-    elif response.headers['Content-Type'] == 'application/json':
-        _print_message_from_json(response)
-    else:
-        print('ERROR: Unexpected content-type of response message')
+        print(f'STATUS: {response.status_code} {http.client.responses[response.status_code]}')
+        print('MESSAGE:')
+        print(f'Concent-Golem-Messages-Version = {response.headers["concent-golem-messages-version"]}')
+        if 'application/octet-stream' in response.headers['Content-Type']:
+            _print_message_from_stream(private_key, public_key, response.content)
+        elif 'application/json' in response.headers['Content-Type']:
+            _print_message_from_json(response)
+        elif 'text/html' in response.headers['Content-Type']:
+            pass
+        else:
+            raise UnexpectedResponse('ERROR: Unexpected content-type of response message')
```

Comment:
This function is meant to just print the response, no matter whether it's correct or now. You should not raise here. Just print what you see. If the response is invalid, you can print that.

If the response is invalid it's `api_request()`'s job to report it.

Reply:
right, done